k8s/helm-chart/dynatrace-gcp-function/templates/active-gate-pod.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Dynatrace LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{- if or (eq .Values.deploymentType "logs") (eq .Values.deploymentType "all") }}
+  {{- if eq .Values.activeGate.useExisting "false" }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: dynatrace-activegate-gcpmon-clusterip
+  namespace: dynatrace
+spec:
+  type: ClusterIP
+  selector:
+    app: dynatrace-activegate-gcpmon
+  ports:
+    - protocol: TCP
+      port: 9999
+      targetPort: 9999
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dynatrace-activegate-gcpmon-pullsecret
+  namespace: dynatrace
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: {{ template "imagePullSecret" . }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: dynatrace-activegate-gcpmon
+  labels:
+    app: dynatrace-activegate-gcpmon
+  namespace: dynatrace
+spec:
+  selector:
+    matchLabels:
+      app: dynatrace-activegate-gcpmon
+  replicas: 1
+  serviceName: dynatrace-activegate-gcpmon
+  template:
+    metadata:
+      labels:
+        app: dynatrace-activegate-gcpmon
+        envid: {{ template "environmentID" . }}
+      annotations:
+        "version": "1"
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+    spec:
+      containers:
+        - name: dynatrace-activegate-gcpmon
+          image: {{ template "activeGateImage" . }}
+          imagePullPolicy: Always
+          ports:
+            - name: ag-port
+              containerPort: 9999
+          env:
+            - name: DT_CAPABILITIES
+              value: "log_analytics_collector"
+            - name: DT_ID_SEED_NAMESPACE
+              value: dynatrace 
+            - name: DT_ID_SEED_PROJECT
+              value: {{ .Values.gcpProjectId }}
+            - name: DT_ID_SEED_ENV_ID
+              value: {{ template "environmentID" . }}
+            - name: DT_ID_SEED_CLUSTER_NAME
+              value: {{ .Values.clusterName }}
+          resources:
+            requests:
+              memory: 1024Mi
+              cpu: 1
+            limits:
+              memory: 1024Mi
```

Comment:
are you sure that request should be the limit? if it is a java then the limit should be at least 20% above

Reply:
why so? if request = limit we have Guaranteed  QoS and this is safe for AG>..

Reply:
if we have request and limit on 1k then java should have around 800Mi because java + OS will use only 1k so if java will take around 1k then k8s will kill pod but java will have no chances to clear garbage collector ex. and work again without restart

Reply:
and also in different way if we are using only 500Mi MAX all the time (not in spikes) so we not need that memory... if our pod want more but not above the limit then k8s will evict other pods to allow our pod to work

Reply:
In Ag repo, in yamls for AG deployment, the request limits are set to 250M. IMHO 1024Mi is more than enough.

Reply:
so limit as 1k is ok or to enough but request should be set to 250 or 500

Reply:
Agreed. Changed to 512MB

Reply:
memory mgmt. is optimized on AG level. AG itself sets heap usage to 50% available limit memory (default java limit is 25%). Having this in mind still we can benefit from Guaranteed QoS.